Bencoding.hpp

```diff
@@ -0,0 +1,56 @@
+// Bencoding.hpp
+#ifndef BENCODING_H
+#define BENCODING_H
+
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <algorithm>
+#include <iterator>
+#include <cctype>
+#include <vector>
+#include <memory>
+#include <cstring>
+#include <unordered_map>
+#include <boost/any.hpp>
+
+class BDecoder {
+private:
+	std::string encodedString;
+	int position;
+	boost::any decodedObject;
+	std::pair<int,int> infoDictPosition;
+
+	bool decodeInteger(boost::any & localOutput);
+	bool decodeString(boost::any & localOutput);
+	bool decodeList(boost::any & localOutput);
+	bool decodeDictionary(boost::any & localOutput);
+	
+public:
+	BDecoder( const std::string &input, int position) {
+		this->encodedString = input;
+		this->position = position;
+	}
+	bool decode(boost::any &localOutput);
+	bool decode();
```

Comment:
The public interface to this class is confusing. What's the interaction between decode(localOutput), decode(), get(), etc? Some comments would help a lot with this.